edit

```diff
@@ -2,58 +2,65 @@
 
 # API documentation: http://www.mediawiki.org/wiki/API:Main_page
 
-. wiki-config
- 
+source wiki-config
+
 TITLE="$1"
 CONTENTS="$2"
 
-if [ -z "$CONTENTS" -o -n "$3" ]; then
-  echo "Syntax: $0 <page_to_edit> <new_content_file>"
+if [ -z "$CONTENTS" ]; then
+  echo "ERROR: Syntax: $0 <page_to_edit> <new_content_file>"
   exit 1
 fi
 
 if [ ! -r "$CONTENTS" ]; then
-  echo "File '$CONTENTS' does not exist"
+  echo "ERROR: File '$CONTENTS' does not exist!"
   exit 2
 fi
 
-echo "Fetching edit token..."
-
-CR=$(curl -s -S \
-	--location \
-	--cookie $cookie_jar \
-	--cookie-jar $cookie_jar \
-	--keepalive-time 60 \
-	--header "Accept-Language: en-GB" \
-	--header "Connection: keep-alive" \
-	--compressed \
-	--request "POST" "${WIKIAPI}?action=tokens&format=txt")
- 
-EDITTOKEN=$(echo "$CR" | awk '/\[edittoken\] =>/ {print $3}')
-
-if [ ${#EDITTOKEN} = 34 ]; then
-	echo "Edit token is: $EDITTOKEN"
-else
-	echo "Edit token not set."
-	echo $CR
-	exit 1
-fi
+echo "INFO: Editing '${WIKIAPI}' '$TITLE'..."
+#echo "INFO: Fetching edit token..."
 
+curl \
+   --silent \
+   --show-error \
+   --location \
+   --cookie "$cookie_jar" \
+   --cookie-jar "$cookie_jar" \
+   --keepalive-time 60 \
+   --header "Accept-Language: en-GB" \
+   --header "Connection: keep-alive" \
+   --compressed \
+   --output "${TMPFOLDER}/fetch-edit-token.json" \
+   --request "GET" \
+   "${WIKIAPI}?action=query&meta=tokens&format=json"
 
-CR=$(curl -S \
-	--location \
-	--cookie $cookie_jar \
-	--cookie-jar $cookie_jar \
-	--keepalive-time 60 \
-	--header "Accept-Language: en-GB" \
-	--header "Connection: keep-alive" \
-	--header "Expect:" \
-	--data-urlencode "title=$TITLE" \
-	--data-urlencode "text@$CONTENTS" \
-	--data-urlencode "token=${EDITTOKEN}" \
-	--request "POST" "${WIKIAPI}?action=edit&format=txt&")
+EDITTOKEN=$(jq --raw-output '.query.tokens.csrftoken' "${TMPFOLDER}/fetch-edit-token.json")
```

Comment:
I am using a temporary file because it's easier to debug manually. The bash xtrace looks gigantic and still cannot trivially be copied into `jq`. Also by keeping all the returned wiki text out, the bash xtrace gets easier to read.